ios/Trifle/Sources/Trifle.swift

```diff
@@ -98,21 +128,37 @@ extension Certificate {
      - returns: true if validated, false otherwise
      */
     public func verify(
-        certificateRequest: MobileCertificateRequest,
-        certificateChain: Array<Certificate>,
-        rootCertificate: Certificate
+        certificateRequest: MobileCertificateRequest?,
+        intermediateChain: Array<Certificate>?,
+        rootCertificate: Certificate?
     ) -> Bool {
-        return X509TrustManager.evaluate([self] + certificateChain + [rootCertificate])
+        
+        // TODO: validate PK in MobileCertificateRequest against certificate
+        var chain : Array<Certificate> = [self]
```

Comment:
this can be:

```
        let chain : Array<Certificate> = [self] + intermediateChain
        if (rootCertificate != nil) {
            chain = [ self ] + chain + [ rootCertificate!]
        } else {
            chain = [ self ] + intermediateChain
        }
```

Reply:
I think you mean
```
        var chain : Array<Certificate> = [self] + intermediateChain
        if (rootCertificate != nil) {
            chain = chain + [ rootCertificate!]
        }

```

Reply:
oops updated. I just prefer immutability

Reply:
I updated although it's not exactly what you asked for. I think it's a good medium.

Reply:
¯\_(ツ)_/¯